### PR TITLE
add python converters to AML coco format

### PR DESCRIPTION
**Pull request by JiaheZhou**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -47,5 +47,7 @@
                                          'vision_tsv_to_iris=vision_datasets.commands.converter_tsv_to_iris:main',
                                          'vision_tsv_to_coco=vision_datasets.commands.converter_tsv_to_coco:main',
                                          'vision_yolo_to_iris=vision_datasets.commands.converter_yolo_darknet_to_iris:main',
-                                         'vision_gen_metafile=vision_datasets.commands.generate_image_meta_info:main']
+                                         'vision_gen_metafile=vision_datasets.commands.generate_image_meta_info:main',
+                                         'vision_IrisCoco_to_AMLCoco=vision_datasets.commands.converter_IrisCoco_to_AMLCoco:main',
```
Comment (pj-ms) [inline setup.py]: lets keep them lower case.
both the command names and file names

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def create_arg_parser():
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Convert IRIS coco to AML coco format.')
+    parser.add_argument('-inputCocoName', '--irisCoco_path', required=True, type=str, help='Iris coco file to convert.')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: try keep the naming style consistent:
- lower case
- separate words by '_', dont use camel case

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    parser.add_argument('-outputJsonName', '--out_put_file', required=True, type=str, default='AML_coco.json', help='output json name.')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: for file path, try to use pathlib

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    parser.add_argument('-imagePath', '--img_path', required=True, type=str, default='images/shelf_images', help='input image path.')
+    parser.add_argument('-absoluteUrl', '--absolute_url', required=True, type=str, default='https://irisdataset.blob.core.windows.net/objectdetection/mobileone_jiahe/images/shelf_images', help='absolute url for blob container.')
+    parser.add_argument('-hasBbox', '--has_bbox', required=True, type=bool, default=True, help='input image path.')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: wrong help message

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def create_arg_parser():
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Convert IRIS coco to AML coco format.')
+    parser.add_argument('-inputCocoName', '--irisCoco_path', required=True, type=str, help='Iris coco file to convert.')
+    parser.add_argument('-outputJsonName', '--out_put_file', required=True, type=str, default='AML_coco.json', help='output json name.')
+    parser.add_argument('-imagePath', '--img_path', required=True, type=str, default='images/shelf_images', help='input image path.')
+    parser.add_argument('-absoluteUrl', '--absolute_url', required=True, type=str, default='https://irisdataset.blob.core.windows.net/objectdetection/mobileone_jiahe/images/shelf_images', help='absolute url for blob container.')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: this is basically telling the world where our storage is

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    parser.add_argument('-hasBbox', '--has_bbox', required=True, type=bool, default=True, help='input image path.')
+
+    parser.add_argument('-projectUrl', '--coco_url', required=False, type=str, default='AmlDatastore://mobileone/', help='project url.')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: unclear what this is

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    parser.add_argument('-dateCaptured', '--date_captured', required=False, type=str, default='2022-06-24T23:34:17.7980622Z', help='date captured.')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: why do we have a default?

Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: what does date captured mean

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+    return parser
+
+def main():
+    args = create_arg_parser().parse_args()
+    logger.info(args.__dict__)
+
+    standard_images = []
+    dir_id_WH = {}
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: pls check naming convention
https://peps.python.org/pep-0008/

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    with open(args.irisCoco_path) as f:
+        iris_file = json.load(f)
+
+        iris_images = iris_file['images']
+        iris_annotations = iris_file["annotations"]
+        iris_categories = iris_file['categories']
+
+        for image in iris_images:
+            standard_tmp = {}
+            WH = []
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: pls check naming convention
https://peps.python.org/pep-0008/

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    standard_coco['images'] = standard_images
+    standard_coco['annotations'] = standard_annotations
+    standard_coco['categories'] = iris_categories
+
+    out_file = open(args.out_put_file, "w")
+    json.dump(standard_coco, out_file, indent=4)
+    out_file.close()
+    
+if __name__ == '__main__':
+    main()
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: empty line at the end of file

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 11]
```diff
@@ -0,0 +1,96 @@
+"""
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: can you run flake8 to check if any warning exists?

File: vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py [hunk 12]
```diff
@@ -0,0 +1,119 @@
+""" 
+target coco format expamle: https://irisdataset.blob.core.windows.net/uvsdatasets/b92_regular_od/train_images.json
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py]: pls fix

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 13]
```diff
@@ -0,0 +1,96 @@
+""" 
+target coco format expamle: https://irisdataset.blob.core.windows.net/uvsdatasets/b92_regular_od/train_images.json
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: pls fix

File: vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+            category['id'] = categories_id
+            category['name'] = label.strip()
+            categories_id += 1
+            categories.append(category)
+
+    AML_coco['images'] = images
+    AML_coco['annotations'] = annotations
+    AML_coco['categories'] = categories
+
+    out_file = open(args.out_put_file, "w")
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py]: with open(...):
   .....

**Review by pj-ms — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (JiaheZhou) [reply setup.py]: Done. Use all lower case

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py (hunk 2 above)
Comment (JiaheZhou) [reply vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: Done

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py (hunk 5 above)
Comment (JiaheZhou) [reply vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: Done, deleted

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py (hunk 11 above)
Comment (JiaheZhou) [reply vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: Done. Passing now

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py (hunk 3 above)
Comment (JiaheZhou) [reply vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: Added path valid check

Comment (pj-ms) [reply vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: example: https://github.com/microsoft/vision-datasets/blob/bdd0ebf5c0c0561486ebb0b96600196b2b89f77c/vision_datasets/commands/convert_local_dir_ic_data.py

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+
+def main():
+    args = create_arg_parser().parse_args()
+    logger.info(args.__dict__)
+
+    standard_images = []
+    dir_id_width_height = {}
+    standard_annotations = []
+
+    if not os.path.exists(args.iris_coco_path):
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: use pathlib

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+logging.basicConfig(level=logging.INFO)
+
+
+def create_arg_parser():
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Convert IRIS coco to AML coco format.')
+    parser.add_argument('-inputCocoName', '--iris_coco_path', required=True, type=str, help='Iris coco file to convert.')
+    parser.add_argument('-outputJsonName', '--out_put_file', required=True, type=str, default='AML_coco.json', help='output json name.')
+    parser.add_argument('-imagePath', '--img_path', required=True, type=str, default='images/shelf_images', help='input image path.')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: avoid camel case for consistency

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+    parser.add_argument('-absoluteUrl', '--absolute_url', required=True, type=str, help='absolute url for blob container.')
+    parser.add_argument('-hasBbox', '--has_bbox', required=True, type=bool, default=True, help='is object dection (has bounding box).')
+    parser.add_argument('-projectPath', '--coco_url', required=False, type=str, default='AmlDatastore://mobileone', help='project path (it is ok to keep default).')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: interesting default.

File: vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+    parser.add_argument('-dateCaptured', '--date_captured', required=False, type=str, default='2022-06-24T23:34:17.7980622Z', help='date captured (it is ok to keep default).')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisCoco_to_AMLCoco.py]: interesting default.

File: vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Convert IRIS coco to AML coco format.')
+    parser.add_argument('-imageTextPath', '--images_txt_path', required=True, type=str, default='test_images.txt', help='image Text file name.')
+    parser.add_argument('-labelTextPath', '--label_name_path', required=True, type=str, default='labels.txt', help='label Text file name.')
+    parser.add_argument('-outputJsonName', '--out_put_file', required=True, type=str, default='AML_coco.json', help='output json name.')
+    parser.add_argument('-absoluteUrl', '--absolute_url', required=True, type=str, default='https://your_image_blob_path', help='absolute url for blob container.')
+    parser.add_argument('-hasBbox', '--has_bbox', required=True, type=bool, default=True, help='is object dection (has bounding box).')
+    parser.add_argument('-projectUrl', '--coco_url', required=False, type=str, default='AmlDatastore://mobileone', help='project path (it is ok to keep default).')
+    parser.add_argument('-dateCaptured', '--date_captured', required=False, type=str, default='2022-06-24T23:34:17.7980622Z', help='date captured (it is ok to keep default).')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py]: Is this needed? What value to put here?

File: vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+def create_arg_parser():
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Convert IRIS coco to AML coco format.')
+    parser.add_argument('-imageTextPath', '--images_txt_path', required=True, type=str, default='test_images.txt', help='image Text file name.')
+    parser.add_argument('-labelTextPath', '--label_name_path', required=True, type=str, default='labels.txt', help='label Text file name.')
+    parser.add_argument('-outputJsonName', '--out_put_file', required=True, type=str, default='AML_coco.json', help='output json name.')
+    parser.add_argument('-absoluteUrl', '--absolute_url', required=True, type=str, default='https://your_image_blob_path', help='absolute url for blob container.')
+    parser.add_argument('-hasBbox', '--has_bbox', required=True, type=bool, default=True, help='is object dection (has bounding box).')
+    parser.add_argument('-projectUrl', '--coco_url', required=False, type=str, default='AmlDatastore://mobileone', help='project path (it is ok to keep default).')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py]: inconsistent naming

File: vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py [hunk 21]
```diff
@@ -0,0 +1,123 @@
...
+logging.basicConfig(level=logging.INFO)
+
+
+def create_arg_parser():
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Convert IRIS coco to AML coco format.')
+    parser.add_argument('-imageTextPath', '--images_txt_path', required=True, type=str, default='test_images.txt', help='image Text file name.')
+    parser.add_argument('-labelTextPath', '--label_name_path', required=True, type=str, default='labels.txt', help='label Text file name.')
+    parser.add_argument('-outputJsonName', '--out_put_file', required=True, type=str, default='AML_coco.json', help='output json name.')
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py]: why are they different

Comment (pj-ms) [reply vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py]: output is a single word

File: vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+        train_images = json.load(f)
+        for line in train_images:
+            image = {}
+            width_height = []
+
+            image_path_zip, label_path_zip = line.strip().split(' ')
+            image_path = '/'.join(image_path_zip.split('.zip@'))
+            label_path = '/'.join(label_path_zip.split('.zip@'))
+
+            img = Image.open(image_path)
```
Comment (pj-ms) [inline vision_datasets/commands/converter_IrisDataset_to_AMLCoco.py]: with xxxx:
    ....

**Review by pj-ms — CHANGES_REQUESTED** [review]

File: setup.py [hunk 23]
```diff
@@ -47,5 +47,7 @@
                                          'vision_tsv_to_iris=vision_datasets.commands.converter_tsv_to_iris:main',
                                          'vision_tsv_to_coco=vision_datasets.commands.converter_tsv_to_coco:main',
                                          'vision_yolo_to_iris=vision_datasets.commands.converter_yolo_darknet_to_iris:main',
-                                         'vision_gen_metafile=vision_datasets.commands.generate_image_meta_info:main']
+                                         'vision_gen_metafile=vision_datasets.commands.generate_image_meta_info:main',
+                                         'vision_iriscoco_to_amlcoco=vision_datasets.commands.converter_iriscoco_to_amlcoco:main',
```
Comment (pj-ms) [inline setup.py]: let us just keep it coco_to_amlcoco (removing iris in the name)

Comment (pj-ms) [reply setup.py]: Can we have a clear doc about what amlcoco requires?

File: setup.py [hunk 24]
```diff
@@ -47,5 +47,7 @@
                                          'vision_tsv_to_iris=vision_datasets.commands.converter_tsv_to_iris:main',
                                          'vision_tsv_to_coco=vision_datasets.commands.converter_tsv_to_coco:main',
                                          'vision_yolo_to_iris=vision_datasets.commands.converter_yolo_darknet_to_iris:main',
-                                         'vision_gen_metafile=vision_datasets.commands.generate_image_meta_info:main']
```
Comment (pj-ms) [inline setup.py:54]: As discussed in our previous meeting, if we consume the data from DatasetHub, we dont have to care about whether the underlying data is iris or coco format

**Review by pj-ms — APPROVED** [review]

**Pull request merged**